### ISSUE TITLE
"Guarantee failed: [!lock->empty()]" on raft branch in RDBBtree.SindexPostConstruct

### ISSUE THREAD

**Issue by danielmewes** [rethinkdb/rethinkdb#4248]
```
[----------] 4 tests from RDBBtree
[ RUN      ] RDBBtree.SindexPostConstruct
Recursively removing directory ./tmp
Previous message may not have been written to the log file (cannot open or find log file).
Version: rethinkdb 2.0.0-1-833-gcc66b7 (debug) (CLANG 3.2 (tags/RELEASE_32/final))
error: Error in ./src/buffer_cache/alt.hpp at line 277:
Previous message may not have been written to the log file (cannot open or find log file).
error: Guarantee failed: [!lock->empty()] 
Previous message may not have been written to the log file (cannot open or find log file).
error: Backtrace:
Previous message may not have been written to the log file (cannot open or find log file).
error: Tue May 19 13:05:36 2015
       
       1: rethinkdb_backtrace(void**, int) at rethinkdb_backtrace.cc:100
       2: backtrace_t::backtrace_t() at backtrace.cc:203
       3: lazy_backtrace_formatter_t::lazy_backtrace_formatter_t() at backtrace.cc:283
       4: format_backtrace(bool) at backtrace.cc:198
       5: report_fatal_error(char const*, int, char const*, ...) at errors.cc:83
       6: buf_parent_t::buf_parent_t(buf_lock_t*) at alt.hpp:277
       7: buf_parent_t::buf_parent_t(buf_lock_t*) at alt.hpp:278
       8: real_superblock_t::expose_buf() at reql_specific.hpp:35
       9: unittest::insert_rows(int, int, store_t*) at rdb_btree.cc:62
       10: unittest::run_RDBBtree_SindexPostConstruct() at rdb_btree.cc:341
       11: std::_Function_handler<void (), void (*)()>::_M_invoke(std::_Any_data const&) at functional:1779
       12: std::function<void ()>::operator()() const at functional:2160
       13: starter_t::run_wrapper(std::function<void ()> const&) at runtime.cc:61
       14: std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)>::operator()(starter_t*, std::function<void ()> const&) const at functional:550
       15: _ZNSt5_BindIFSt7_Mem_fnIM9starter_tFvRKSt8functionIFvvEEEEPS1_S4_EE6__callIvJEJLi0ELi1EEEET_OSt5tupleIJDpT0_EESt12_Index_tupleIJXspT1_EEE+0x70 at functional:1145
       16: _ZNSt5_BindIFSt7_Mem_fnIM9starter_tFvRKSt8functionIFvvEEEEPS1_S4_EEclIJEvEET0_DpOT_+0x26 at functional:1203
       17: std::_Function_handler<void (), std::_Bind<std::_Mem_fn<void (starter_t::*)(std::function<void ()> const&)> (starter_t*, std::function<void ()>)> >::_M_invoke(std::_Any_data const&) at functional:1779
       18: std::function<void ()>::operator()() const at functional:2160
       19: callable_action_instance_t<std::function<void ()>&>::run_action() at callable_action.hpp:31
       20: callable_action_wrapper_t::run() at runtime_utils.cc:43
       21: coro_t::run() at coroutines.cc:205
       22: coro_t* coro_t::spawn_sometime<std::function<void ()>&>(std::function<void ()>&&&) at coroutines.hpp:58
       23: starter_t::on_thread_switch() at runtime.cc:57
       24: linux_message_hub_t::on_event(int) at message_hub.cc:154
       25: epoll_event_queue_t::run() at epoll.cc:115
       26: linux_thread_pool_t::start_thread(void*) at thread_pool.cc:155
       27: /lib/x86_64-linux-gnu/libpthread.so.0(+0x7e9a) [0x7fa78ade9e9a] at 0x7fa78ade9e9a (/lib/x86_64-linux-gnu/libpthread.so.0)
       28: clone+0x6d at 0x7fa78ab168bd (/lib/x86_64-linux-gnu/libc.so.6)
Previous message may not have been written to the log file (cannot open or find log file).
error: Exiting.
Previous message may not have been written to the log file (cannot open or find log file).
Trace/breakpoint trap (core dumped)
```

**Comment by timmaxw** [comment]
In CR 2946.

**Comment by timmaxw** [comment]
Merged in faccf8a458f8ac58fdfdbb57e668f674b64facde.
